app/models/study.rb

```diff
@@ -1354,6 +1354,19 @@ def all_expression_matrix_cells
     all_cells.uniq # account for raw counts & processed matrix files repeating cell names
   end
 
+  # array of all cell names from non-raw count matrix files
+  # used in filtering distribution-based expression visualizations (violin/dot plots)
+  def processed_matrix_cells
```

Comment:
it's not strictly related to visualization, but I would consider moving this method to `expression_viz_service`.  I could see it going either way